src/gl/init.c

```diff
@@ -14,7 +14,15 @@
 #include "debug.h"
 #include "loader.h"
 #include "logs.h"
+
+#define fpe_state_t struct fpe_state_s
```

Comment:
why do you need that? can't you `typedef struct fpe_state_s fpe_state_t` instead?

Reply:
`typedef struct struct_name_s struct_name_t` gets tricky for C99 with multiple cross includes. GCC produces lots of 'type redefinition` errors if it encounter two identical strings like that from multiple includes. So the solution is to use defines. However to add minimal changes for existing codebase I undef such defines in same header or source file immediately after usage.

Reply:
To show the problem consider such code:
_header1.h_:
...
`#include "header2.h"`
...
`typedef struct struct_name_s struct_name_t;`
_header2.h_:
...
`#include "header1.h"`
...
`typedef struct struct_name_s struct_name_t;`

so the result for any of includes in source code would lead to error:
`typedef struct fpe_state_s fpe_state_t;` - from header 1 or header 2
`typedef struct fpe_state_s fpe_state_t;` - vice versa: from header 2 or header 1
gcc error: 'type redefinition `fpe_state_t` already defined in...'

my solution is define guard:
_header1.h_:
...
`#include "header2.h"`
...
`#ifndef struct_name_t`
`typedef struct struct_name_s struct_name_t;`
`#endif`
_header2.h_:
...
`#include "header1.h"`
...
`#ifndef struct_name_t`
`typedef struct struct_name_s struct_name_t;`
`#endif`
and use this internaly like:
```cpp
#include "header1.h"
#define struct_name_t struct_name_t // or define struct_name_t as struct struct_name_s
#include "header2.h"
#undef struct_name_t
```

Reply:
I don't have any error on my side as long as the typedef is consistant across files.

Reply:
Mmm, so it seems it's a C11 extension. typedef redefined is forbidden in C99.

Reply:
Why not using C11 as default then? Every compiler should be compatible now?

Reply:
And then to complicate it even more there comes cross includes :
_header1.h_:
...
`#include "header2.h"`
...
`typedef struct struct_name_s struct_name_t;`
_header2.h_:
...
`#include "header1.h"`
...
`typedef struct struct_name_s struct_name_t;`

so the result for any of includes in source code
```cpp
#include "header2.h"
```
would lead to same error:
`typedef struct struct_name_s struct_name_t;` - from header 1 or header 2
`typedef struct struct_name_s struct_name_t;` - vice versa: from header 2 or header 1
gcc error: 'type redefinition `struct_name_t` already defined in...'

my solution is use define guards localy in header:
_header1.h_:
...
`#include "some_header_that_includes_header2.h"`
...
`#ifdef _HEADER2_INC_ // there is typedef already`
`#   ifndef struct_name_t`
`#   define struct_name_t struct_name_t`
`#   define DO_NOT_FORGET_TO_UNDEF_struct_name_t`
`#endif`
...
`#ifndef struct_name_t`
`typedef struct struct_name_s struct_name_t;`
`#endif`
...
`#ifdef DO_NOT_FORGET_TO_UNDEF_struct_name_t`
`#   undef struct_name_t`
`#   undef DO_NOT_FORGET_TO_UNDEF_struct_name_t`
`#endif`
_header2.h_:
...
`#include "some_header_that_includes_header1.h"`
...
`#ifdef _HEADER1_INC_ // there is typedef already`
`#   ifndef struct_name_t`
`#   define struct_name_t struct_name_t`
`#   define DO_NOT_FORGET_TO_UNDEF_struct_name_t`
`#endif`
...
`#ifndef struct_name_t`
`typedef struct struct_name_s struct_name_t;`
`#endif`
...
`#ifdef DO_NOT_FORGET_TO_UNDEF_struct_name_t`
`#   undef struct_name_t`
`#   undef DO_NOT_FORGET_TO_UNDEF_struct_name_t`
`#endif`

Reply:
> Mmm, so it seems it's a C11 extension. typedef redefined is forbidden in C99.

yep it is.



> Why not using C11 as default then? Every compiler should be compatible now?

Not every compiler but C99 compatible compiler should be. For me the usage case is some quite old embedded platform with no C11 and library compiles just fine with 20-30 string changed like in this Pull Request. I hope you don't mind accepting changes to satisfy C99 requirements.

Reply:
If there is an actual use case, then I don't mind.

Reply:
Many thanks for library and I will add hopefully small Pullrequests to work around C99 limitations in a couple of hours.